pkg/daemon/daemon.go

```diff
@@ -1999,7 +2003,13 @@ func (dn *Daemon) checkStateOnFirstRun() error {
 	}
 	// currentConfig != desiredConfig, and we're not booting up into the desiredConfig.
 	// Kick off an update.
-	return dn.triggerUpdate(state.currentConfig, state.desiredConfig, state.currentImage, state.desiredImage)
+	err = dn.triggerUpdate(state.currentConfig, state.desiredConfig, state.currentImage, state.desiredImage)
+	if err != nil && apierrors.IsNotFound(err) {
+		// if this is first run, the MC we went to get DNE this means there was a content mismatch
+		// bwtn what user provided and what was rendered. State that here.
+		return fmt.Errorf("User provided configuration caused a mismatch in the rendered config. Please compare /etc/mcs-machine-config-content.json and the existing rendered config. %w", err)
```

Comment:
So, this error here actually triggers on the curentConfig, meaning that we don't get to this line if we have this issue. 

I think you would need to add it to almost the beginning of the function, in `getStateAndConfigs` return if you want this. Maybe a better phrasing would be:

```
Could not find config %v in-cluster, this likely indicates the MachineConfigs in-cluster has changed during the install process. If you are seeing this when installing the cluster, please compare the in-cluster rendered machineconfigs to ...
```

or is that a bit too wordy? You are theoretically able to hit this later on if you delete a config while the node is rebooting, for example, so I don't want to confuse the user